### DISCUSSION TITLE
Thai translation is not working as expected

### DISCUSSION THREAD

**Discussion by EugeneGohh** [ant-design/ant-design#40263] [Q&A]
Hey everyone! I'm facing a problem in using [Ant Design Internationalization](https://4x.ant.design/docs/react/i18n).

I've followed the exact setup as said in the documentation. The languages I'm using in my app are included in **supported languages** which are English `en_US`,  Korean `ko_KR` & Thai `th_TH`.

However, only English `en_US` & Korean `ko_KR` are working, **but not for Thai** `th_TH`.

Below is my code.

**Package JSON**
`package.json`

`"antd": "^4.24.7",`

**App.tsx**
`/src/app.tsx`

```
import ko_kR from 'antd/es/locale/ko_KR';
import en_US from 'antd/es/locale/en_US';
import th_TH from 'antd/es/locale/th_TH';
import { getLocale } from 'umi';
const currentLocale = getLocale();

const AppProvider = ({ children, routes }: any) => {
  const queryClient = new QueryClient();

  const newChildren = React.cloneElement(children, {
    ...children.props,
    routes,
  });

  return (
    <Suspense fallback={<PageLoading />}>
      <ConfigProvider
        locale={currentLocale === 'ko-KR' ? ko_kR : currentLocale === 'th-TH' ? th_TH : en_US}
      >
        <QueryClientProvider client={queryClient}>{newChildren}</QueryClientProvider>
      </ConfigProvider>
    </Suspense>
  );
};

export function rootContainer(container: any) {
  return React.createElement(AppProvider, null, container);
}
```

**Images Below**

When my current locale is **Korean**, it's translating.
<img width="178" alt="image" src="https://user-images.githubusercontent.com/64187129/212628838-d8455c73-46f1-48de-94a4-5bc2292c901c.png">

When my current locale is **English**, it's translating.
<img width="207" alt="image" src="https://user-images.githubusercontent.com/64187129/212628891-735143b9-c6c0-4138-bc18-6726ed3f4c4e.png">

When my current locale is **Thai**, it's not translating.
<img width="182" alt="image" src="https://user-images.githubusercontent.com/64187129/212628659-3a362351-447c-4870-b065-91b934d2e8cb.png">

**Comment by zombieJ** [comment]
Localization is from social contribution. If find that some locale not translated, please feel free to patch for PR : )

**Reply by EugeneGohh** [reply]
The locale file exists in that path, but then the translation is not translating as expected

<img width="638" alt="image" src="https://user-images.githubusercontent.com/64187129/212630449-22585065-3737-4901-97c4-04353a33ef94.png">

**Reply by yoyo837** [reply]
Please open an issue with a reproduction for this.

**Reply by EugeneGohh** [reply]
> Please open an issue with a reproduction for this.

Is it necessary to open an issue? It's working in the sandbox.

**Comment by yoyo837** [comment]
<img width="1070" alt="image" src="https://user-images.githubusercontent.com/6134547/212636518-ec97a99a-1e74-420d-914e-f67b71d88b2b.png">
https://codesandbox.io/s/shai-xuan-he-pai-xu-antd-5-1-5-forked-ul2ibf?file=/index.tsx

**Reply by EugeneGohh** [reply]
Thanks for the update! I've tried Codesandbox link you sent over & it's working but unfortunately, it's not working on my side. Any idea why is this happening?

**Reply by EugeneGohh** [reply]
@yoyo837 

Notice the difference here?
![image](https://user-images.githubusercontent.com/64187129/212646075-b523e454-68dd-4b4b-8577-8ad947cc422a.png)

**Reply by EugeneGohh** [reply]
cc: @zombieJ

**Reply by EugeneGohh** [reply]
Kindly check out this repo in Codesandbox.

https://codesandbox.io/s/guo-ji-hua-xiang-guan-de-pei-zhi-forked-072lfz?file=/App.tsx

**Reply by yoyo837** [reply]
@chenshuai2144

**Reply by zombieJ** [reply]
I search the related issue, use `ConfigProviderWrap` can pass locale: https://github.com/ant-design/pro-components/issues/4962

But found that Thai translate is not full filled. I guess you can PR pro-components to fill missing translation.

**Reply by EugeneGohh** [reply]
Where is the entry point to **pro-components** to fill missing translation?

> I search the related issue, use `ConfigProviderWrap` can pass locale: [ant-design/pro-components#4962](https://github.com/ant-design/pro-components/issues/4962)
> 
> But found that Thai translate is not full filled. I guess you can PR pro-components to fill missing translation.

**Reply by EugeneGohh** [reply]
I tested out the solution left in the issue, but it's not working on my site either. 

Wondering is this the entry point to fill in?
https://github.com/ant-design/pro-components

@zombieJ

**Comment by EugeneGohh** [comment]
Update: Posted on **[Ant Design - Pro Components Repository](https://github.com/ant-design/pro-components/issues/6529)**

cc: @zombieJ @yoyo837 @chenshuai2144

**Comment by afc163** [comment]
> I guess you can PR pro-components to fill missing translation.

It is better to create a PR instead of creating another issue that we already aware of.

**Reply by EugeneGohh** [reply]
May I know where is the entry point/starting point to fix this? Or any PRs/issues I can refer to?

**Reply by EugeneGohh** [reply]
The pull request is up: https://github.com/ant-design/pro-components/pull/6535

**cc**: @zombieJ @yoyo837 @chenshuai2144

**Comment by EugeneGohh** [comment]
My PR has been merged but after I upgraded the version from `"@ant-design/pro-components": "^2.3.8"` to `"@ant-design/pro-components": "2.3.57"`, it seems like the changes are not reflecting on my side. The below picture attached is what I'm receiving.

Are you able to assist me on this on where to make the changes in the repository?

<img width="282" alt="image" src="https://user-images.githubusercontent.com/64187129/221506354-1f4c0b52-58ae-425b-921d-487a8e8e9e6b.png">

@zombieJ @yoyo837 @chenshuai2144

**Reply by EugeneGohh** [reply]
Reference: https://procomponents.ant.design/en-US/components/table?current=1&pageSize=5#internationalization-related-configuration

**Comment by craftedsro** [comment]
I have same problem with sk_SK and cs_CZ
